### PR TITLE
feat: support non-ascii characters in `for/each` title placeholders

### PR DESCRIPTION
**Pull request by k-yle**
### Description

although it's unusual, it's theoretically possible to use non-ascii characters in variable names. For example:

```js

// math-heavy code might use greek letters:
it.each`
  ω    | Σ
  ${1} | ${5}
  ${2} | ${10}
`('returns true when $ω is < $Σ', ({ ω, Σ }) => {
  // ...
});


// and I suppose this works in some other languages too:
it.each`
  时间戳 | 结果
  ${1}  | ${5}
  ${2}  | ${10}
`('returns $结果 given $时间戳', ({ 结果, 时间戳 }) => {
  // ...
});
```

currently, the test title formatter use RegEx `\w`, so it doesn't support non-ascii values. [`\p{ID_Continue}`](https://www.unicode.org/reports/tr31/#D1) has been [supported since node v10](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escape), see Unicode's [UAX31-D1](https://www.unicode.org/reports/tr31/#D1) for more info.

<table>
<tr>
 <td>Before
 <td>After
<tr>
 <td>image
 <td>image
</table>
 

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.
- [x] Please check [Allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to make review process faster. Note that this option is not available for repositories that are owned by Github organizations.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] ~~If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.~~

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: test/unit/test/each.test.ts [hunk 1]
```diff
@@ -225,6 +225,14 @@ ${{ val: 3 }}   | ${'b'} | ${'3b'}
   expect(a.val + b).toBe(expected)
 })
 
+test.each`
```
Comment (k-yle) [inline test/unit/test/each.test.ts:228]: the tests don't actually prove that this change works, because there is no way to write tests for `formatTitle` at the moment

Comment (hi-ogawa) [reply test/unit/test/each.test.ts:228]: There should be some in test/e2e and some reporter. Can you check?

Comment (k-yle) [reply test/unit/test/each.test.ts:228]: thanks, i missed that earlier. added an e2e test now

**Review by hi-ogawa — APPROVED** [review]
Looks good to me.

I've let agents do some adversarial edge cases and it surfaced interesting case. Some looks technically breaking change-ish, but probably fine (we land this in next v5 anyways).

| Placeholder | Before: `\w` | After: `\p{ID_Continue}` |
|---|---|---|
| `$对象.结果` | No match; remains literal | `row["对象"]["结果"]` |
| `$a.结果` | Captures `a.` → `row.a[""] + "结果"` | `row.a["结果"]` |
| `$𐐀` | No match | `row["𐐀"]` |
| `$é` | No match | `row["é"]` |
| `$é` | Captures `e` → `row.e + "\u0301"` | `row["e\u0301"]` |
| `$fooé` | Captures `foo` → `row.foo + "é"` | `row["fooé"]` |
| `$a‌b` | Captures `a` → `row.a + "\u200Cb"` | `row["a\u200Cb"]` |
| `$a·b` | Captures `a` → `row.a + "·b"` | `row["a·b"]` |
| `$ａｂｃ` | No match | `row["ａｂｃ"]` |
| `$́foo` | No match | `row["\u0301foo"]` |
| `$foo-bar` | `row.foo + "-bar"` | Unchanged |
| `$💡` | Remains literal | Unchanged |
| `$.foo` | `row[""]["foo"]` | Unchanged |
| `$foo..bar` | `row.foo[""].bar` | Unchanged |
| `$0` | `items[0]` | Unchanged |

Also side note, it looks like people normally use it like `[$_\p{ID_Start}][$\p{ID_Continue}]*` to match javascript identifier-ish string, but in our case, we do want to match non identifier for array placeholder like `0` (in title `$0`), so just matching `ID_Continue` seems like a good choice.

**Review by sheremet-va — APPROVED** [review]
